cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcer.java

```diff
@@ -43,12 +48,23 @@
 
   private final AuthorizerInstantiator authorizerInstantiator;
   private final boolean propagatePrivileges;
+  private final Principal masterUser;
 
   @Inject
   DefaultAuthorizationEnforcer(CConfiguration cConf, AuthorizerInstantiator authorizerInstantiator) {
     super(cConf);
     this.propagatePrivileges = cConf.getBoolean(Constants.Security.Authorization.PROPAGATE_PRIVILEGES);
     this.authorizerInstantiator = authorizerInstantiator;
+    Principal master;
+    try {
+      // the master user will be the one who runs the cdap
+      master = new Principal(UserGroupInformation.getLoginUser().getShortUserName(), Principal.PrincipalType.USER);
+    } catch (IOException e) {
+      LOG.warn("Failed to get the login user", e);
+      // if failed to get the login user, get the master principal instead
```

Comment:
I think we should propagate the exception instead of falling back to the master principal from the cConf.
If we're unable to call `UserGroupInformation.getLoginUser().getShortUserName()`, then I think that should be fatal.